ECH/ECK remote clusters doc is incorrect for trust.yml (#1647)
elastic/docs-content: deploy-manage/remote-clusters/ec-enable-ccs-for-eck.md

# Remote clusters between {{ech}} and ECK [ec-enable-ccs-for-eck]

These steps describe how to configure remote clusters between an {{es}} cluster in {{ech}} (ECH) and an {{es}} cluster running within [{{eck}} (ECK)](/deploy-manage/deploy/cloud-on-k8s.md). Once that’s done, you’ll be able to [run CCS queries from {{es}}](/solutions/search/cross-cluster-search.md) or [set up CCR](/deploy-manage/tools/cross-cluster-replication/set-up-cross-cluster-replication.md).

> ## Establish trust between two clusters [ec_establish_trust_between_two_clusters]

Reviewer: ```suggestion
## Establish trust between the two clusters [ec_establish_trust_between_two_clusters]
```

    2. Select **Certificates** as authentication mechanism and click **Next**.

    3. In **Add trusted CA certificate**, upload the `eck-ca.crt` file retrieved in the previous step.

    4. In **Select trusted clusters**, configure the following:
        * Select **Trust clusters whose Common Name follows the Elastic pattern**.
        * For **Scope ID**, enter `<kubernetes-namespace>.es.local`, replacing `<kubernetes-namespace>` with the namespace of your ECK cluster.
        * In **Trust**, select **All deployments**.

    5. In **Name the environment**, enter a name for the trusted environment. That name will appear in the trust summary of your deployment’s **Security** page.

    6. Select **Create trust** to complete the configuration.

    7. On the confirmation screen, when prompted **Have you already set up trust from the other environment?**, select **No, I have NOT set up trust from the other environment yet**. Download both the ECH deployment CA certificate and the `trust.yml` file. These files can also be retrieved from the **Security** page of the deployment. You’ll use these files to configure trust in the ECK deployment.

> ### Establish trust in the ECK cluster [ec_establish_trust_in_the_eck_cluster]

Reviewer: :2c: I'd split out editing trust.yml into its own procedure because it's too long for a single step

1. Edit the `trust.yml` file downloaded in the previous step from the Cloud UI to adapt it to your ECK cluster. The file includes a subject name pattern that is not compatible with ECK.

>     Replace the line corresponding to the `Scope ID` you entered when configuring trust in the ECH deployment:
> 
>     ```sh
>     "*.node.*.cluster.<kubernetes-namespace>.es.local.account"
>     ```
> 
>     with the correct subject name for your ECK cluster, using the following pattern:
> 
>     ```sh
>     "*.node.<cluster-name>.<kubernetes-namespace>.es.local"
>     ```

Reviewer: avoid sentence fragments like this

```suggestion
    Replace the line corresponding to the `Scope ID` you entered when configuring trust in the ECH deployment: 

    ```sh
    "*.node.*.cluster.<kubernetes-namespace>.es.local.account"
    ```

    Replace it with the correct subject name for your ECK cluster. The new subject name should use the following pattern:

    ```sh
    "*.node.<cluster-name>.<kubernetes-namespace>.es.local"
    ```
```

    ::::{important}
    If you don’t update this entry, {{es}} nodes of your ECK deployment may fail to start or join the cluster due to failed trust validation.
    ::::

>     For example, the original downloaded file might contain:

Reviewer: ```suggestion
    For example, the original downloaded file might contain the following:
```

    ```yaml
    trust.subject_name:
      - "*.node.2dc556bb4bd040e00d0135683b66a2f6.cluster.1075999151.account" <1>
      - "*.node.*.cluster.<kubernetes-namespace>.es.local.account" <2>
    ```
    1. This entry identifies your ECH deployment. Leave it unchanged.
    2. This entry identifies your ECK deployment incorrectly, and must be updated.

>     For an ECK cluster named `quickstart` in the `default` namespace, the updated file should look like:

Reviewer: ```suggestion
    For an ECK cluster named `quickstart` in the `default` namespace, the updated file should look like the following:
```

    ```yaml
    trust.subject_name:
      - "*.node.2dc556bb4bd040e89d0135683b66a2f6.cluster.1075708151.account"
      - "*.node.quickstart.default.es.local"
    ```

    Apply the changes and save the `trust.yml` file.

2. In the same namespace as your {{es}} cluster, upload the ECH CA certificate that you downloaded from the Cloud UI as a Kubernetes Secret:

    ```sh
    kubectl create secret generic remote-ech-ca --from-file=ca.crt=<path-to-CA-certificate-file> -n <namespace>
    ```

> 3. In the same namespace as your {{es}} cluster, upload the updated `trust.yml` file as a Kubernetes ConfigMap. For a cluster named `quickstart`, run:

Reviewer: ```suggestion
3. In the same namespace as your {{es}} cluster, upload the updated `trust.yml` file as a Kubernetes ConfigMap. For a cluster named `quickstart`, run the following command:
```

    ```sh
    kubectl create configmap quickstart-trust-ech --from-file=trust.yml=<path-to-trust.yml> -n <namespace>
    ```

> 4. Edit the {{es}} Kubernetes resource to include the following configuration. This example assumes that the Kubernetes secret and config map created in the previous steps are named `remote-ech-ca` and `quickstart-trust-ech`, respectively:

Reviewer: ```suggestion
4. Edit the {{es}} Kubernetes resource to reference the new certificate and trust.yml file. This example assumes that the Kubernetes secret and ConfigMap created in the previous steps are named `remote-ech-ca` and `quickstart-trust-ech`, respectively:
```

    ::::{note}

>     Apply these changes to all `nodeSets` of your cluster. Updating this configuration will restart all {{es}} pods, which may take some time to complete.

Reviewer: ```suggestion
    Apply these changes to all `nodeSets` of your cluster. Updating this configuration will restart all {{es}} pods, which might take some time to complete.
```
